admin/tabs/activity/clienterrors/ClientErrorDetail.js

```diff
@@ -4,79 +4,69 @@
  *
  * Copyright © 2019 Extremely Heavy Industries Inc.
  */
-import {Component} from 'react';
 import {dialog} from '@xh/hoist/kit/blueprint';
-import {HoistComponent, elemFactory} from '@xh/hoist/core';
+import {hoistComponent, useProvidedModel} from '@xh/hoist/core';
 import {filler, table, tbody, tr, th, td} from '@xh/hoist/cmp/layout';
 import {clipboardButton} from '@xh/hoist/desktop/cmp/clipboard';
 import {jsonInput} from '@xh/hoist/desktop/cmp/input';
 import {toolbar} from '@xh/hoist/desktop/cmp/toolbar';
 import {button} from '@xh/hoist/desktop/cmp/button';
 import {fmtDateTime} from '@xh/hoist/format';
+import {ClientErrorModel} from './ClientErrorModel';
 
-@HoistComponent
-export class ClientErrorDetail extends Component {
+export const [ClientErrorDetail, clientErrorDetail] = hoistComponent(props => {
+    const model = useProvidedModel(ClientErrorModel, props),
+        rec = model.detailRecord;
 
-    render() {
-        const {model} =  this,
-            rec = model.detailRecord;
+    if (!rec) return null;
 
-        if (!rec) return null;
+    return dialog({
+        title: 'Error Details',
+        style: {width: 1000},
+        isOpen: true,
+        onClose: () => model.closeDetail(),
+        items: renderDetail(model, rec)
+    });
+});
 
-        return dialog({
-            title: 'Error Details',
-            style: {width: 1000},
-            isOpen: true,
-            onClose: this.onCloseClick,
-            items: this.renderDetail(rec)
-        });
-    }
-
-    renderDetail(rec) {
-        return [
-            table({
-                className: 'xh-admin-error-detail',
-                items: [
-                    tbody(
-                        tr(th('User:'), td(rec.username)),
-                        tr(th('Message:'), td(rec.msg || 'None provided')),
-                        tr(th('User Alerted:'), td(`${rec.userAlerted}`)),
-                        tr(th('Device/Browser:'), td(`${rec.device}/${rec.browser}`)),
-                        tr(th('Agent:'), td(rec.userAgent)),
-                        tr(th('App Version:'), td(rec.appVersion)),
-                        tr(th('Environment:'), td(rec.appEnvironment)),
-                        tr(th('Date:'), td(fmtDateTime(rec.dateCreated)))
-                    )
-                ]
-            }),
-            jsonInput({
-                value: rec.error,
-                disabled: true,
-                height: 450,
-                width: '100%',
-                editorProps: {lineWrapping: true}
+//----------------
+// Implementation
+//----------------
```

Comment:
Trivial point, but I am thinking we could lose all these Implementation block comments where we are building up a component from non-exported functions.

Reply:
Agree